### ISSUE TITLE
How to set batchsize for maDLC

### ISSUE THREAD

**Issue by DorianBattivelli** [DeepLabCut/DeepLabCut#1182]
Hello,

I'm still facing a crashing issue for training on a ma-project on Windows 10, using the 2.1.10.2 dlc version.
After you answered to my last issue (#1173), the reduction of batch size + cropping for training worked. Now I added some frames to retrain the model and colab crashes again when I run the training. The strange fact is that although I define "batch size = 1" in the config file, I still see that it is = 8 when I run the training cell on colab. And so, I still face the resource exhausted error. 

<img width="418" alt="Schermata 2021-04-08 alle 08 23 06" src="https://user-images.githubusercontent.com/66886884/113978010-b44b5a00-9843-11eb-8d2f-dd13e403dd0a.png">

What could I do to fix this issue? 
Thank you, 
Best,
Dorian

**Comment by MMathisLab** [comment]
The `config.yaml` sets batchsize for video analysis only; for training, you need to either edit the `pose_cfg.yaml` file. You should also pass `allow_growth = True`.

For every function, if you look at the docstrings it will give you all the options. For example:

```python
ipython
import deeplabcut
deeplabcut.train_network?

train_network(config, shuffle=1, trainingsetindex=0, max_snapshots_to_keep=5, displayiters=None, saveiters=None, maxiters=None, allow_growth=False, gputouse=None, autotune=False, keepdeconvweights=True, modelprefix='')
    Trains the network with the labels in the training dataset.

    Parameter
    ----------
    config : string
        Full path of the config.yaml file as a string.

    shuffle: int, optional
        Integer value specifying the shuffle index to select for training. Default is set to 1

    trainingsetindex: int, optional
        Integer specifying which TrainingsetFraction to use. By default the first (note that TrainingFraction is a list in config.yaml).

    Additional parameters:

    max_snapshots_to_keep: int, or None. Sets how many snapshots are kept, i.e. states of the trained network. Every savinginteration many times
        a snapshot is stored, however only the last max_snapshots_to_keep many are kept! If you change this to None, then all are kept.
        See: https://github.com/AlexEMG/DeepLabCut/issues/8#issuecomment-387404835

    displayiters: this variable is actually set in pose_config.yaml. However, you can overwrite it with this hack. Don't use this regularly, just if you are too lazy to dig out
        the pose_config.yaml file for the corresponding project. If None, the value from there is used, otherwise it is overwritten! Default: None

    saveiters: this variable is actually set in pose_config.yaml. However, you can overwrite it with this hack. Don't use this regularly, just if you are too lazy to dig out
        the pose_config.yaml file for the corresponding project. If None, the value from there is used, otherwise it is overwritten! Default: None

    maxiters: this variable is actually set in pose_config.yaml. However, you can overwrite it with this hack. Don't use this regularly, just if you are too lazy to dig out
        the pose_config.yaml file for the corresponding project. If None, the value from there is used, otherwise it is overwritten! Default: None

    allow_groth: bool, default false.
        For some smaller GPUs the memory issues happen. If true, the memory allocator does not pre-allocate the entire specified
        GPU memory region, instead starting small and growing as needed. See issue: https://forum.image.sc/t/how-to-stop-running-out-of-vram/30551/2

    gputouse: int, optional. Natural number indicating the number of your GPU (see number in nvidia-smi). If you do not have a GPU put None.
        See: https://nvidia.custhelp.com/app/answers/detail/a_id/3751/~/useful-nvidia-smi-queries

    autotune: property of TensorFlow, somehow faster if 'false' (as Eldar found out, see https://github.com/tensorflow/tensorflow/issues/13317). Default: False
:

**Comment by MMathisLab** [comment]
And, if you change the batchsize, you need to carefully think about how long to train for; i.e. the default 50K would not be enough unless you do batchsize 8, which is what we recommend! 

My guess is your new data is not cropped to the same size. You can't use batch training unless all images match dimensions.... so, you need to crop the new frames.

**Comment by DorianBattivelli** [comment]
Hello @MMathisLab ,

Thank you for the support. 
Ok so, if I go for batch size = 1, I should do 50000  x 8 iterations: 400 000 iterations for training? 

For the cropping, if you talk about the cropimagesandlabels function, I also cropped the new frames with 600x600 as I did for the first set of labeled frames. Should I do something else?

**Comment by DorianBattivelli** [comment]
Everything works now, thank you very much, 

Best, 
Dorian
